CMake/resolve_dependency_modules/folly/CMakeLists.txt

```
@@ -25,6 +25,16 @@ resolve_dependency_url(FOLLY)
 
 message(STATUS "Building Folly from source")
 
+# Required so cmake can use the correct version of the OpenSSL dependency for
+# MacOS
+if(APPLE)
+  execute_process(
+    COMMAND brew --prefix openssl@1.1
+    OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  set(ENV{OPENSSL_ROOT_DIR} ${OPENSSL_ROOT_DIR})
+endif()
```

We should give the user some feedback (and stop the build) if openssl wasn't found. And also only do this if the user hasn't already set the var. (written in the webgui please check and format locally^^)
```suggestion
if(APPLE AND NOT DEFINED OPENSSL_ROOT_DIR)
  find_program(BREW brew)
  if(BREW)
    execute_process(
      COMMAND brew --prefix openssl@1.1
      OUTPUT_VARIABLE OPENSSL_ROOT_DIR
      OUTPUT_STRIP_TRAILING_WHITESPACE)
    set(ENV{OPENSSL_ROOT_DIR} ${OPENSSL_ROOT_DIR})
  endif()
  if(NOT OPENSSL_ROOT_DIR)
    message(FATAL_ERROR "Unable to locate brew prefix for OpenSSL 1.1. Please install with `brew install openssl@1.1` or set OPENSSL_ROOT_DIR"
  endif()
endif()
```

---

Please ping after you have done this @rishitc  and I will start the merge process...

---

Hi @kgpai @assignUser,
Thanks for the feedback! I'll start working on the changes and revert 👍 

Thanks.